assets/src/edit-story/components/helpCenter/useHelpCenter.js

```diff
@@ -102,6 +157,26 @@ export function useHelpCenter() {
     [store.actions]
   );
 
+  useEffect(() => {
+    let isMounted = true;
+    getCurrentUser()
+      .then(({ meta }) => {
+        isMounted &&
+          actions.hydrateReadTipsSuccess({
+            readTips: meta.web_stories_onboarding ?? {},
+          });
+      })
+      .catch(actions.persitingReadTipsError);
+    return () => (isMounted = false);
+  }, [actions, getCurrentUser]);
+
+  const persistenceKey = createKeyFromBooleanMap(store.state.readTips);
+  useEffect(() => {
+    updateCurrentUser({
+      meta: { web_stories_onboarding: createBooleanMapFromKey(persistenceKey) },
+    }).catch(actions.persitingReadTipsError);
+  }, [actions, updateCurrentUser, persistenceKey]);
```

Comment:
The `updateCurrentUser` logic should be moved the currentUser provider and state should be updated of the current User.